### DISCUSSION TITLE
Predictions for Time Series of Binomial Distributed Values

### DISCUSSION THREAD

**Discussion by deejlucas** [awslabs/gluonts#1217] [Q&A]
My data:
Several million weekly-frequency time series, some going back decades. These are counts of activities that are strictly limited to two per week. Most of these time series are constant zero after some run of generally non-zero values. Attrition is not constant (the probability of a given observation being the last non-zero observation in a time series decreases with time but not constantly). This confounds models used in this area that assume constant attrition. 

Given that a value is non-zero, it is much more likely to be two than one. This makes the negative binomial distribution, often used for DeepAR with count targets, fit the data poorly.

It seems to me that this data is distributed according to either a zero-inflated binomial distribution (a "quitting" process produces zero with some probability, and time series that persist (e.g. don't quit) have some probability of incrementing from 0 to 1 and 2) or a beta-binomial distribution (say something like n=2, a=0.1, b=0.25).

My predictions:
The next t (4 or more) observations for several hundred thousand of the time series above (those with recent non-zero values). I would like to implement a DeepAR model, as we believe there are elements of seasonality, serial correlation, and interactions across wide time gaps.

My question:
I would consider trying to implement one of the distributions that seems right to me (zero-inflated binomial and/or beta-binomial), but I don't want to waste my time if this is not worth doing. Is there an implemented distribution that is likely to work in this case or approximates either or both of these distributions? Or is there some technical issue that prevents these distributions from being implemented?

**Comment by lostella** [comment]
@deejlucas regarding binomial (zero inflation one should be able to get for free, similarly to [the ZINB case](https://github.com/awslabs/gluon-ts/blob/4a836a003f41f53b35c613874de404cf2baf6cfa/src/gluonts/mx/distribution/neg_binomial.py#L141)) I think the biggest difficulty might be in having `n` be output by a model, since it’s a natural number. But maybe one can fix that at model definition time? At least for your use case, that should be OK (`n=2`)

**Reply by timoschowski** [reply]
Maybe another thing to try here is the categorical distribution:
https://ts.gluon.ai/api/gluonts/gluonts.distribution.categorical.html

It should be easy to try out and maybe it achieves what you want?
